WordPress/Classes/ViewRelated/Jetpack/Install/ViewModel/WPComJetpackRemoteInstallViewModel.swift

```diff
@@ -0,0 +1,102 @@
+/// Controls the Jetpack Remote Install flow for Jetpack-connected self-hosted sites.
+///
+/// A site can establish a Jetpack connection through individual Jetpack plugins, but the site may not have
+/// the full Jetpack plugin. This covers the logic behind the plugin installation process, and will stop the
+/// process before proceeding to the Jetpack connection step (since the site is already connected).
+///
+class WPComJetpackRemoteInstallViewModel {
+
+    // MARK: Dependencies
+
+    private let service: PluginJetpackProxyService
+
+    // MARK: Properties
+
+    // The flow should always complete after the plugin is installed.
+    let shouldConnectToJetpack = false
+
+    var onChangeState: ((JetpackRemoteInstallState, JetpackRemoteInstallStateViewData) -> Void)? = nil
+
+    private(set) var state: JetpackRemoteInstallState = .install {
+        didSet {
+            onChangeState?(state, viewData)
+        }
+    }
+
+    // MARK: Methods
+
+    init(service: PluginJetpackProxyService = .init()) {
+        self.service = service
+    }
+}
+
+// MARK: - View Model Implementation
+
+extension WPComJetpackRemoteInstallViewModel: JetpackRemoteInstallViewModel {
+    func viewReady() {
+        // set the initial state & trigger the callback.
+        state = .install
+    }
+
+    func installJetpack(for blog: Blog, isRetry: Bool) {
+        // Ensure that the blog is accessible through a WP.com account,
+        // and doesn't already have the Jetpack plugin.
+        guard let siteID = blog.dotComID?.intValue,
+              blog.jetpackIsConnectedWithoutFullPlugin else {
+            // In this case, let's do nothing for now. Falling to this state should be a logic error.
+            return
```

Comment:
Just to confirm, it shouldn't be possible to hit this state, correct? Otherwise the button does nothing.

Reply:
Yes, it shouldn't be possible... But technically it's possible through logic/implementation error from our side.